src/cli_support.ts

```diff
@@ -10,13 +10,17 @@ import * as path from 'path';
 
 // Postprocess generated JS.
 export function pathToModuleName(context: string, fileName: string): string {
-  fileName = fileName.replace(/\.js$/, '');
+  fileName = fileName.replace(/\.[tj]s$/, '');
 
   if (fileName[0] === '.') {
     // './foo' or '../foo'.
     // Resolve the path against the dirname of the current module.
     fileName = path.join(path.dirname(context), fileName);
   }
+
+  // Ensure consistency by naming all modules after their absolute paths
+  fileName = path.resolve(fileName);
```

Comment:
This is indeed a problem, and your fix is pretty close I think.

We cannot quite use absolute path names I think, because then you'll end up with non-reproducible names across different developer machines, build systems, and so on.

Maybe you could try to first create an absolute path, and then relativize against the project's source roots?

Reply:
Sorry for the delay on this -- I've been sidetracked by other projects. This sounds like a better idea, and as a bonus all the module names will be easier to read. I'll make this change either tomorrow or Monday.

Reply:
@mprobst I've updated this. I had to make it a bit more complex than just resolving against the `sourceRoot` property since it's possible to specify files on the command line.